### PR TITLE
Migrate `link-cfg` and `rustdoc-default-output` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

try-job: aarch64-apple
try-job: x86_64-msvc
try-job: x86_64-mingw
try-job: x86_64-gnu-llvm-18
try-job: i686-msvc

### CODE REVIEW

File: tests/run-make/rustdoc-default-output/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+use run_make_support::{bare_rustdoc, diff};
+
+fn main() {
+    let out = bare_rustdoc().run().stdout_utf8();
+    diff()
+        .expected_file("output-default.stdout")
+        .actual_text("actual", out)
+        // replace the channel type in the URL with $CHANNEL
+        .normalize(r"/nightly/|beta/|stable/|1\.[0-9]+\.[0-9]+/", "$$CHANNEL/")
```
Comment (jieyouxu) [inline tests/run-make/rustdoc-default-output/rmake.rs:14]: Problem: maybe missing a `/` somewhere, is it supposed to be `/$CHANNEL/`?

**Comment by Oneirical** [comment]
@rustbot ready

The slash fix worked!

**Review by jieyouxu — APPROVED** [review]
Thanks!

**Comment by Oneirical** [comment]
Only `nightly` has the extra slash in the pattern, which made the test look successful when it actually only worked on nightly builds. I have deleted the leading slash, this should work now.

@bors r=@jieyouxu

**Pull request merged**